subsys/bluetooth/controller/ll_sw/lll.h

```diff
@@ -281,9 +281,16 @@ struct node_rx_hdr {
 	};
 };
 
+/* Template node rx type with memory aligned offset to PDU buffer.
+ * NOTE: offset to memory aligned pdu buffer location is used to reference
+ *       node rx type specific information, like, terminate or sync lost reason
+ *       from a dedicated node rx structure storage location.
+ */
 struct node_rx_pdu {
 	struct node_rx_hdr hdr;
-	uint8_t               pdu[0];
+	union {
+		uint8_t    pdu[0] __aligned(4);
```

Comment:
I'm not sure I understand why the `__aligned(4)` part is necessary. Wouldn't the the `pdu[0]` fields be aligned correctly without it?

Reply:
`pdu[0]` is uint8_t and hence depending on the previous structure member size, would be placed unaligned (packed). HCI code typecasts unpacked structures to `&pdu[0]`, potentially leading to unaligned access faults if `struct node_rx_hdr` size changes.

Reply:
Alternatively, the below too solves the problem.

```suggestion
	union {
		uint8_t    pdu[0];
		uint32_t   param[0];
	};
```

Reply:
I see. Wouldn't defining pdu as a `void *` provide the same? Do we lose anything from doing that? 

It's not an issue that it's there, but I guess we'll be casting `pdu` to another struct definition anyhow, so does it matter that it's uint8_t or void?

Reply:
Defining pdu as void* would always occupy 4 bytes, even when not needed.

I think having the zero-length array with `__aligned(4)` is the correct solution. I am not sure whether the union in the `node_rx_pdu` struct is strictly necessary, but it is consistent with how the other struct look like, which is a good thing.

Reply:
You are correct :) It could be `uint32_t pdu[0]` though, but I think that `__aligned(4)` serves the same purpose just as well.